modules/cloud-run-v2/variables.tf

```diff
@@ -125,6 +125,41 @@ variable "iam" {
   default     = {}
 }
 
+variable "iap_config" {
+  description = "If present, turns on Identity-Aware Proxy (IAP) for the Cloud Run service."
+  type = object({
+    iam          = optional(list(string))
+    iam_additive = optional(list(string))
+  })
+  default = null
+
+  validation {
+    condition = var.iap_config == null || (
+      (try(var.iap_config.iam, null) != null && try(var.iap_config.iam_additive, null) == null) ||
+      (try(var.iap_config.iam, null) == null && try(var.iap_config.iam_additive, null) != null)
+    )
+    error_message = "When iap_config is provided, exactly one of 'iam' or 'iam_additive' must be specified."
```

Comment:
This is not needed. We want to allow providing `iap_config = {}` (just to enable IAP, but not do any IAM on service)